src/main/java/com/github/veluvexiau/umldoc/classfile/ClassFileParser.java

```diff
@@ -0,0 +1,173 @@
+
+  /**
+   * Read a jar file given by the parameter Path and return a list of Entity.
+   *
+   * @param path File's path
+   * @return List of Entity List created by the method containing all entities
+   * @throws IOException if the file cannot be opened
+   */
+  public static List<Entity> readJarFile(Path path) throws IOException {
+    var entities = new ArrayList<Entity>();
+    var finder = ModuleFinder.of(path);
+    for (var moduleReference : finder.findAll()) {
+      try (var reader = moduleReference.open()) {
+        for (var filename : (Iterable<String>) reader.list()::iterator) {
+          if (!filename.endsWith(".class")) {
+            continue;
+          }
+          try (var inputStream = reader.open(filename).orElseThrow()) {
+            var classReader = new ClassReader(inputStream);
+            classReader.accept(new ClassVisitor(Opcodes.ASM9) {
+
+              private static Modifier modifier(int access) {
+                if (java.lang.reflect.Modifier.isPublic(access)) {
+                  return Modifier.PUBLIC;
+                }
+                if (java.lang.reflect.Modifier.isPrivate(access)) {
+                  return Modifier.PRIVATE;
+                }
+                if (java.lang.reflect.Modifier.isProtected(access)) {
+                  return Modifier.PROTECTED;
+                }
+                return null;
+                //throw new IllegalStateException("Modifier not recognized" + access);
+              }
+
+              @Override
+              public void visit(int vers,
+                                int access,
+                                String name,
+                                String sign,
+                                String supName,
+                                String[] inter) {
+                Entity entity;
+                if (java.lang.reflect.Modifier.isInterface(access)) {
+                  entity = new Entity(Set.of(),
+                      new TypeInfo(Optional.empty(), name, List.of()),
+                      Stereotype.INTERFACE,
+                      List.of(),
+                      List.of());
+                } else if ((access & Opcodes.ACC_RECORD) != 0) {
+                  entity = new Entity(Set.of(),
+                      new TypeInfo(Optional.empty(), name, List.of()),
+                      Stereotype.RECORD,
+                      List.of(),
+                      List.of());
+                } else if ((access & Opcodes.ACC_ENUM) != 0) {
+                  entity = new Entity(Set.of(),
+                      new TypeInfo(Optional.empty(), name, List.of()),
+                      Stereotype.ENUM,
+                      List.of(),
+                      List.of());
+                } else {
+                  entity = new Entity(Set.of(),
+                      new TypeInfo(Optional.empty(), name, List.of()),
+                      Stereotype.CLASS,
+                      List.of(),
+                      List.of());
+                }
```

Comment:
too many code duplication, refactor the code to have a method that takes the access and returns the stereotype